### PR TITLE
<Tags/> - missing highlight when typing (autocomplete feature)

### PR DESCRIPTION
**Pull request by Faradey27**
### What changed

- fixed <Tags/> highlight when typing (autocomplete feature)

### Why it changed

- it was broken

---

- [x] Change is tested
- [x] Change is documented
- [x] Build is green
- [ ] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

**Comment by Faradey27** [comment]
@shlomitc

**Comment by Faradey27** [comment]
@shlomitc by default highlight prop is setted to false, change it to true and re-check

without my changes - nothing
with them - highlight works

**Comment by shlomitc** [comment]
I understand, so you mean passing `highlight={true}` prop will allow it. This is nice but there are two issues:
1. This should be the default behavior
2. When `highlight` is on -> the tags options become textual and lose any icon/tag they have (take a look at the Arizona option for example)

**Comment by Faradey27** [comment]
@shlomitc 
1) fixed
2) good catch, re-worked algoritm to support this

File: src/Highlighter/Highlighter.js [hunk 1]
```diff
@@ -3,6 +3,58 @@ import WixComponent from '../BaseComponents/WixComponent';
...
 import PropTypes from 'prop-types';
 
+const childKeyGenerator = () => {
+  let childKey = 0;
+  return () => `highlighted-child-${childKey++}`;
+};
+
+let nextChildKey = childKeyGenerator();
+
+const _isReactElement = element => element && element.props !== undefined;
```
Comment (hepiyellow) [inline src/Highlighter/Highlighter.js]: While we're at it. Please use `React.isValidElement()`.
(double check me if this is really the required validation you need)

File: src/Highlighter/Highlighter.js [hunk 2]
```diff
@@ -16,91 +68,16 @@ class Highlighter extends WixComponent {
...
-      }
-      case this._CHILD_TYPES.ARRAY:
-        return this._renderArrayOfChildren(element, parent);
-      default:
-        return;
-    }
+  constructor() {
+    super();
+    // we want to drop counter on component creation
+    nextChildKey = childKeyGenerator();
```
Comment (hepiyellow) [inline src/Highlighter/Highlighter.js]: why don't we reset the counter on constructor?
Does it overflow nicely (if it will ever happen)...?

**Comment by Faradey27** [comment]
@hepiyellow @shlomitc updated, build is green

**Comment by shlomitc** [comment]
Looks good.
@wuwa - highlighting is now default for the Tags component, please verify so we can merge

**Comment by wuwa** [comment]
cool, looks good

**Pull request merged**